src/printer.js

```diff
@@ -3859,6 +3859,10 @@ function makeString(rawContent, enclosingQuote) {
   // Matches _any_ escape and unescaped quotes (both single and double).
   const regex = /\\([\s\S])|(['"])/g;
 
+  // Matches any unnecessarily escaped character.
+  // Adapted from https://github.com/eslint/eslint/blob/de0b4ad7bd820ade41b1f606008bea68683dc11a/lib/rules/no-useless-escape.js#L27
+  const regexUnnecessaryStringEscapes = /(^|[^\\])\\([^\\nrvtbfux\r\n\u2028\u2029"'0])/;
```

Comment:
@lydell or @bakkot, could you review this piece of code? I'm very scared that this may introduce bugs.